### DISCUSSION TITLE
Consolidating all custom imports in one place

### DISCUSSION THREAD

**Discussion by pa-0** [moudey/Shell#542] [Q&A] [answered]
I'm trying to figure out the best approach to adding new custom items, for example, to the "File Manage" and "Develop" menus in a way that won't require me to edit "develop.nss" and "file-manage.nss" (by adding the necessary "`import 'imports/customdev.nss'` in "develop.nss" and "`import 'imports/newfmitems.nss'`" in "file-manage.nss" respectively) everytime the app is updated.  

Is it possible to consolidate all of these changes into one .nss file that gets imported directly into "shell.nss"?   Or alternatively, is there a better workflow to applying these changes to each new version of Shell?

**Comment by pa-0** [comment]
Also is it possible to add a new item to the File Manage menu <ins>without</ins> modifying `file-manage.nss`?

**Reply by ghost** [reply, marked as answer]
Yes, you can as long as the item or menu you want to add is not a part/inside of another menu.
```js
item(title='Test' menu='File Manage')
```
Nested menus are separated with `/`. `File Manage/Copy Path`

**Reply by pa-0** [reply]
Thank you!

**Reply by pa-0** [reply]
Would you mind providing a simple example for creating a new menu inside `File Manage` that doesn't need to be placed in `file-manage.nss` (similar to the one you provided for a new item)?

**Reply by pa-0** [reply]
This is what I have so far, but I'm guessing it's not the correct syntax unless I actually placed it inside of `folder-manage.nss`:
```nwscript
menu(where=sel.count>0 type='file|dir|drive|namespace|back' mode="multiple" title='File Manage' image=\uE253)
{
    menu(mode="single" type='back' expanded=true)
    {
        menu(separator="before" title='Add to Folder' image=icon.new_folder)
        {
            item(title='New Folder with Selection' menu = 'Add to Folder' type='file|dir' mode='multiple' image = \ue290 cmd-line='/c set "newFolder=New folder with selection" && call md "%newFolder%" && for %a in (@sel) do call move "%a" "%newFolder%"' window='hidden')
            item(title='New Folder "@sel.title" with Selection' menu = 'Add to Folder' type='file|dir' mode='multiple' cmd-line='/c set "newFolder=@sel.title" && call md "%newFolder%" && for %a in (@sel) do call move "%a" "%newFolder%"' window='hidden')
        }
        
    }
}

```

**Reply by ghost** [reply]
something similar, 
```js
menu (title='Folder Stuff...' menu='File Manage'){
	item(title='Folder with Selection' type='file|dir' mode='multiple' icon=icon.move_to
    	commands {
        	cmd=if(input("Directory Name", "Enter your directory name:") & len(input.result)>0, io.directory.create(input.result)),
        	cmd=io.move(sel.path,path.combine( sel.parent, input.result, sel.name)) & command.refresh invoke=1 // io.move messes up on desktop
    	})
	item(title="Move to existing Folder" type='file|dir' image=icon.move_to mode='multiple' 
		commands {
    		cmd=input("Directory Name",'Enter Folder Name:'),
    		cmd=io.move(sel.path, path.combine(sel.parent,input.result,sel.name)) invoke=1
    	})
}
```

**Reply by pa-0** [reply]
Ahh ok.  Thanks again for your help!

**Comment by RubicBG** [comment]
> Тhe easiest way to redirect nss files: Create folder "Nilesoft Shell" in "C:\", copy "shell.nss" and "import" directory to newly created folder "C:\Nilesoft Shell", open "C:\Program Files\Nilesoft Shell\shell.nss", delete everything in it and add only this line:
> 
> import 'C:\Nilesoft Shell\shell.nss'

**Reply by pa-0** [reply]
thanks yes, I found that in the dicussions and am currently using it.  I just wasn't sure if it were possible (after moving the contents of that folder over to a custom directory) to leave those moved files unchanged while still customizing the menus created within them, in new .nss files.  I think the commenter above answered my question though.
